tests/components/concentration_tracker/test_concentration_tracker_for_diffusion.py

```diff
@@ -0,0 +1,281 @@
+    ]
+
+    link_check = np.array([0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
+
+    node_check = np.array([1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0])
+
+    np.testing.assert_equal(link_field, link_check)
+
+    for node_field in node_fields:
+        np.testing.assert_equal(node_field, node_check)
+
+
+# Test that user input of grid fields produces the correct fields
+def test_fields_for_user_field_input():
+    mg = RasterModelGrid((3, 3))
+    _ = mg.add_zeros("soil__flux", at="link")
+    _ = mg.add_zeros("soil__depth", at="node")
+    _ = mg.add_zeros("topographic__elevation", at="node")
+    _ = mg.add_zeros("soil_production__rate", at="node")
+
+    c_sed = mg.add_ones("sediment_property__concentration", at="node")
+    c_br = mg.add_ones("bedrock_property__concentration", at="node")
+    p = mg.add_ones("sediment_property_production__rate", at="node")
+    d = mg.add_ones("sediment_property_decay__rate", at="node")
+
+    _ = ConcentrationTrackerForDiffusion(
+        mg,
+        concentration_initial=c_sed,
+        concentration_in_bedrock=c_br,
+        local_production_rate=p,
+        local_decay_rate=d,
+    )
+
+    link_field = mg.at_link["sediment_property__mass_flux"]
+    node_fields = [
+        mg.at_node["sediment_property__concentration"],
+        mg.at_node["bedrock_property__concentration"],
+        mg.at_node["sediment_property_production__rate"],
+        mg.at_node["sediment_property_decay__rate"],
+    ]
+
+    link_check = np.array([0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
+
+    node_check = np.array([1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0])
+
+    np.testing.assert_equal(link_field, link_check)
+
+    for node_field in node_fields:
+        np.testing.assert_equal(node_field, node_check)
+
+
+# Test that physically impossible inputs raise correct errors
+def test_properties_concentrations():
+    """
+    ConcentrationTrackerForDiffusion should throw an error when input
+    concentration values are negative.
+    """
+    mg = RasterModelGrid((3, 3))
+    _ = mg.add_zeros("soil__flux", at="link")
+    _ = mg.add_zeros("soil__depth", at="node")
+    _ = mg.add_zeros("topographic__elevation", at="node")
+    _ = mg.add_zeros("soil_production__rate", at="node")
+
+    # Instantiate the component
+    with pytest.raises(ValueError):
+        _ = ConcentrationTrackerForDiffusion(mg, concentration_initial=-1)
+    # Instantiate the component
+    with pytest.raises(ValueError):
+        _ = ConcentrationTrackerForDiffusion(mg, concentration_in_bedrock=-1)
+
+
+# %% Test against analytical solutions
+# PLACEHOLDER: Test results against 1-D analytical solution (for DepthDependentDiffuser)
+# (I think this is covered by the docstring tests, so I haven't added it here)
+
+# PLACEHOLDER: Test results against 1-D analytical solution (for DepthDependentTaylorDiffuser)
+# (I think this is covered by the docstring tests, so I haven't added it here)
```

Comment:
These are great ideas! I would like to see more tests that show the component is working correctly. Another test to include would be one that ensures mass balance.